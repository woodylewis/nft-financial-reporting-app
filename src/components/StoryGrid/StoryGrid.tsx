import * as React from 'react';
import PropTypes from 'prop-types';

import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';

type MetaDataProp = {
  networkValue: object;
};

const StoryGrid = (prop: MetaDataProp) => {
  const { networkValue } = prop;

  return (
    <Grid container spacing={2}>
      {Object.entries(networkValue).map((cell) => (
        <Grid item key={cell[0]} xs={12} md={4}>
          <Typography
            sx={{
              color: '000',
              backgroundColor: '#c8e6c9',
              border: '.05rem solid #999',
              height: '2rem',
              textAlign: 'left',
              verticalAlign: 'text-top',
              fontWeight: 700,
              fontSize: 14,
            }}
            pt={1}
            px={2}
            borderRadius={1}
          >
            {cell[0]}: {cell[1]}
          </Typography>
        </Grid>
      ))}
    </Grid>
  );
};

StoryGrid.propTypes = {
  networkValue: PropTypes.object.isRequired,
};

export default StoryGrid;
