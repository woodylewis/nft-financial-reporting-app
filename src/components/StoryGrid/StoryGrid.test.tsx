import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import StoryGrid from './StoryGrid';

describe('StoryGrid component', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const networkValue = {};
    const { asFragment } = render(<StoryGrid networkValue={networkValue} />);
    expect(asFragment).toMatchSnapshot();
  });
});
