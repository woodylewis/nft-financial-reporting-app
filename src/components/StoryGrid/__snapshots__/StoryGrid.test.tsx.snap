// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoryGrid component should match snapshot 1`] = `[Function]`;
